### DISCUSSION TITLE
How to add initial displacement or velocity on specific material points in Peridigm?

### DISCUSSION THREAD

**Discussion by guangdongzhang** [peridigm/peridigm#236] [Q&A]
How to add initial displacement or velocity on specific material points in Peridigm? And how to output displacement or velocity on specific material points. What's the command in the Peridigm file ".yaml".

**Comment by oldninja** [comment]
Hi, you can look at the example files:
  Boundary Conditions: 
       Node Set One: "node_set_1.txt" 
       Node Set Two: "node_set_2.txt" 
       Bar Initial Velocity: 
           Type: "Initial Velocity" 
           Node Set: "Node Set One" 
           Coordinate: "x" 
           Value: "100.0" 
       Bar End Fixed Displacement X: 
           Type: "Prescribed Displacement" 
           Node Set: "Node Set Two" 
           Coordinate: "x" 
           Value: "0.0" 
       Bar End Fixed Displacement Y: 
           Type: "Prescribed Displacement" 
           Node Set: "Node Set Two" 
           Coordinate: "y" 
           Value: "0.0" 
       Bar End Fixed Displacement Z: 
           Type: "Prescribed Displacement" 
           Node Set: "Node Set Two" 
           Coordinate: "z" 
           Value: "0.0"

**Reply by guangdongzhang** [reply]
Hello, how to correctly export a mesh file ".g" from Cubit?

**Comment by guangdongzhang** [comment]
Yes, does it mean I should set several material points as  node sets where I put initial displacement field on the sets?
How should I output the displacement on specific material point when the calculation is finished?


| |
张广栋
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 7/20/2022 10:21 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [peridigm/peridigm] How to add initial displacement or velocity on specific material points in Peridigm? (Discussion #236) |

Hi, you can look at the example files:
  Boundary Conditions:
      Node Set One: "node_set_1.txt"
      Node Set Two: "node_set_2.txt"
      Bar Initial Velocity:
          Type: "Initial Velocity"
          Node Set: "Node Set One"
          Coordinate: "x"
          Value: "100.0"
      Bar End Fixed Displacement X:
          Type: "Prescribed Displacement"
          Node Set: "Node Set Two"
          Coordinate: "x"
          Value: "0.0"
      Bar End Fixed Displacement Y:
          Type: "Prescribed Displacement"
          Node Set: "Node Set Two"
          Coordinate: "y"
          Value: "0.0"
      Bar End Fixed Displacement Z:
          Type: "Prescribed Displacement"
          Node Set: "Node Set Two"
          Coordinate: "z"
          Value: "0.0"

**Reply by oldninja** [reply]
Yes, that is the correct procedure: define node sets to apply boundary conditions (displacements/velocities/forces). 
For the output, you can request what you need. In general will be for the entire model. 

Look at the example folder, there are good examples that are easy to follow and will answer these questions.

**Comment by guangdongzhang** [comment]
For the output results, if I am interested in one direction displacement. How should I set the output, for example x , y, or z direction displacement. Does the software ParaView work for this requirement?


| |
张广栋
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 7/20/2022 11:51 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [peridigm/peridigm] How to add initial displacement or velocity on specific material points in Peridigm? (Discussion #236) |

Yes, that is the correct procedure: define node sets to apply boundary conditions (displacements/velocities/forces).
For the output, you can request what you need. In general will be for the entire model.

Look at the example folder, there are good examples that are easy to follow and will answer these questions.

**Reply by oldninja** [reply]
ParaView.
